modules/icgc-argo-workflows/payload/qcmetrics/resources/usr/bin/main.py

```diff
@@ -0,0 +1,272 @@
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  Authors:
+    Linda Xiang
+"""
+
+import os
+import sys
+import argparse
+import subprocess
+import json
+import re
+import hashlib
+import uuid
+import tarfile
+from datetime import date
+import copy
+from glob import glob
+import yaml
+import csv
+import io
+
+workflow_process_map = {
+    'Pre Alignment QC': 'pre_aln'
+}
+
+tool_list = ['fastqc', 'cutadapt', 'multiqc']
+
+def calculate_size(file_path):
+    return os.stat(file_path).st_size
+
+
+def calculate_md5(file_path):
+    md5 = hashlib.md5()
+    with open(file_path, 'rb') as f:
+        for chunk in iter(lambda: f.read(1024 * 1024), b''):
+            md5.update(chunk)
+    return md5.hexdigest()
+
+def get_files_info(file_to_upload, date_str, analysis_dict, process_indicator):
+    file_info = {
+        'fileSize': calculate_size(file_to_upload),
+        'fileMd5sum': calculate_md5(file_to_upload),
+        'fileAccess': 'controlled',
+        'info': {
+            'data_category': 'Quality Control Metrics',
+            'data_subtypes': None,
+            'files_in_tgz': []
+        }
+    }
+
+    if re.match(r'.+?fastqc\.tgz$', file_to_upload):
+        file_type = 'fastqc'
+        file_info.update({'dataType': 'Sequencing QC'})
+        file_info['info']['data_subtypes'] = ['Sample Metrics']
+        file_info['info'].update({'analysis_tools': ['FastQC']})
+        file_info['info'].update({'description': 'High level sequencing reads QC metrics generated by FastQC.'})
+
+    elif re.match(r'.+?cutadapt\.tgz$', file_to_upload):
+        file_type = 'cutadapt'
+        file_info.update({'dataType': 'Sequencing QC'})
+        file_info['info']['data_subtypes'] = ['Sample Metrics']
+        file_info['info'].update({'analysis_tools': ['Cutadapt']})
+        file_info['info'].update({'description': 'Perform cutadpat on sequencing reads and generate log file.'})
+
+    elif re.match(r'.+?multiqc\.tgz$', file_to_upload):
+        file_type = 'multiqc'
+        file_info.update({'dataType': 'Sequencing QC'})
+        file_info['info']['data_subtypes'] = ['Sample Metrics']
+        file_info['info'].update({'analysis_tools': ['MultiQC']})
+
+        # retrieve qc metrics from multiqc_data
+        metric_info = get_mqc_general_stats(file_to_upload)
+        file_info['info'].update({'metrics': metric_info})
```

Comment:
Is `fastqc` and `cutadapt` missing `'info':'metrics'`?

Reply:
only `multiqc` will have `info: metric` field.